reasoner/resolution/resolver/RootResolver.java

```diff
@@ -169,16 +170,16 @@ protected void receiveExhausted(Response.Exhausted fromDownstream, int iteration
 
     @Override
     protected void initialiseDownstreamActors() {
-        Set<Concludable> concludablesWithApplicableRules = Iterators.iterate(concludables)
-                .filter(c -> c.getApplicableRules(conceptMgr, logicMgr).hasNext()).toSet();
-        Set<Retrievable> retrievables = Retrievable.extractFrom(conjunction, concludablesWithApplicableRules);
-        Set<Resolvable> resolvables = new HashSet<>();
-        resolvables.addAll(concludablesWithApplicableRules);
-        resolvables.addAll(retrievables);
-        plan = planner.plan(resolvables);
-        iterate(plan).forEachRemaining(resolvable -> {
-            downstreamResolvers.put(resolvable, registry.registerResolvable(resolvable));
-        });
+        if (!concludables.isEmpty()) {
+            Set<Retrievable> retrievables = Retrievable.extractFrom(conjunction, concludables);
+            Set<Resolvable> resolvables = new HashSet<>();
+            resolvables.addAll(concludables);
+            resolvables.addAll(retrievables);
+            plan.addAll(planner.plan(resolvables));
+            iterate(plan).forEachRemaining(resolvable -> {
+                downstreamResolvers.put(resolvable, registry.registerResolvable(resolvable));
+            });
+        }
```

Comment:
We should use the same strategy in `RuleResolver`!

Reply:
done!